### PR TITLE
fix(database-permission): could not call inside dbms package in SQL console

### LINKED ISSUE #2391: [Bug]: failed to call dbms package in oracle mode
### ODC version

4.2.4

### OB version

not related

### What happened?

type a dbms calling statement and execute it:
截屏2024-05-15 11 25 58

### What did you expect to happen?

the package starts with dbms is the internal package in oracle mode, we shouldn't stop them~ and there no work around method to execute them.

### How can we reproduce it (as minimally and precisely as possible)?

follow what happened~

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by smallsheeeep**
#### What type of PR is this?
type-bug
module-database permission

#### What this PR does / why we need it:
When execute:
```sql
call DBMS_OUTPUT.putline('aaaa');
```
We cannot judge `DBMS_OUTPUT` is name of a database or a package. Now we will recognize them as database name, but the "database" is not exist, so the SQL will be intercepted during permission check.
This PR fix it by add a whitelist, the default vaule is all OB inside system packages. User can add items by config metaDB. If a new dialect type is supported, we no need to change codes, but add config key as `odc.permission-check.whitelist.database.[dialect-type]`.

#### Which issue(s) this PR fixes:
Fixes #2391 

#### Special notes for your reviewer:
Self-test passed.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-server/src/main/resources/data.sql [hunk 1]
```diff
@@ -810,3 +810,10 @@ INSERT INTO config_system_configuration(`key`, `value`, `description`) VALUES('o
  'mysql, information_schema, test', 'schema exclusions when synchronizing MySQL database schema') ON DUPLICATE KEY UPDATE `id`=`id`;
 INSERT INTO config_system_configuration(`key`, `value`, `description`) VALUES('odc.database.schema.sync.exclude-schemas.doris',
  'mysql, information_schema, test', 'schema exclusions when synchronizing MySQL database schema') ON DUPLICATE KEY UPDATE `id`=`id`;
+
+INSERT INTO config_system_configuration(`key`, `value`, `description`) VALUES('odc.permission-check.whitelist.database.ob-mysql',
+ 'DBMS_RESOURCE_MANAGER, DBMS_STATS, DBMS_UDR, DBMS_XPLAN, DBMS_WORKLOAD_REPOSITORY',
+ 'Permission check whitelist for database in OB MySQL dialect type') ON DUPLICATE KEY UPDATE `id`=`id`;
+INSERT INTO config_system_configuration(`key`, `value`, `description`) VALUES('odc.permission-check.whitelist.database.ob-oracle',
```
Comment (yizhouxw) [inline server/odc-server/src/main/resources/data.sql:817]: may collect from oceanbase v4.3.0 via
```sql
select OBJECT_NAME from ALL_OBJECTS  where OWNER='SYS' AND OBJECT_TYPE='PACKAGE' order by OBJECT_NAME;
```

File: server/odc-service/src/main/java/com/oceanbase/odc/service/permission/common/PermissionCheckWhitelist.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * @author gaoda.xy
+ * @date 2024/5/16 15:38
+ */
+@Data
+@RefreshScope
+@Configuration
+@ConfigurationProperties(prefix = "odc.permission-check.whitelist")
+public class PermissionCheckWhitelist {
+
+    private Map<String, List<String>> database;
```
[the remark this answers is not in the document]
Comment (yizhouxw) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/permission/common/PermissionCheckWhitelist.java:42]: build Set via `new TreeSet<>(String.CASE_INSENSITIVE_ORDER)` for case insensitive match

[the remark this answers is not in the document]
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/permission/common/PermissionCheckWhitelist.java]: I use `@PostConstruct` to construct `TreeSet` for better performance.

File: server/odc-server/src/main/resources/data.sql (hunk 1 above)
Comment (smallsheeeep) [reply server/odc-server/src/main/resources/data.sql]: You are right, I modified it.

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**